### ISSUE TITLE
List assignment test with mismatched number of elements incorrectly passes

### ISSUE THREAD

**Issue by denis-bogdanas** [ethereum/vyper#703]
As the title says.

https://github.com/ethereum/vyper/blob/3eb961ecccb8ced76f19217843bb9ee88f6f8750/tests/parser/features/iteration/test_for_in_list.py#L8

**Comment by BenKaufman10** [comment]
I am pretty sure the elements count is completely fine, remember that arrays start for 0 so 0 to 5 is actually 6 elements (here s[0] is 1, s[1] is 2, and so on).

**Comment by denis-bogdanas** [comment]
This looks contradictory with the syntax of common languages. I expected `n` in `int128[n]` to mean the number of elements, like it is in C/Java/etc.

**Comment by denis-bogdanas** [comment]
In fact the check is not performed at all. Try `s: int128[1] = [1, 2, 3, 4, 5, 6] ` .

**Comment by fubuloubu** [comment]
Yup, this is a bug. I think a bug in the parsing section of the compiler. If you try to set a global with that variable `s`, it fails right?

**Comment by denis-bogdanas** [comment]
Yes, if `s` is global assignment fails with `Mismatched number of elements.`

**Comment by BenKaufman10** [comment]
I see this bug is pretty wide. This code also works:
`struct: {a: int128, b: decimal} = {a: 1.2, b: 1} `
And fails when struct is a global variable.

**Comment by fubuloubu** [comment]
I think those seems pretty similar. Sounds like we have our work cut out for us!

**Comment by jacqueswww** [comment]
@denis-bogdanas thanks for the report! :smile:
